pkg/runtime/depot.go

```diff
@@ -253,10 +278,28 @@ func (d *depot) DeployViaCopy(id strfmt.UUID, relativeSrc, absoluteDest string)
 		Files:       files.RelativePaths(),
 		RelativeSrc: relativeSrc,
 	})
+	d.recordUse(id)
 
 	return nil
 }
 
+func (d *depot) recordUse(id strfmt.UUID) {
+	// Ensure a cache entry for this artifact exists and then update its last access time.
+	if _, exists := d.config.Cache[id]; !exists {
+		size, err := fileutils.GetDirSize(d.Path(id))
+		if err != nil {
+			multilog.Error("Could not get artifact size on disk: %v", err)
+			size = 0
+		}
+		logging.Debug("Recording artifact '%s' with size %.1f MB", id.String(), float64(size)/float64(MB))
+		d.config.Cache[id] = &artifactInfo{Size: size}
+	} else {
+		logging.Debug("Recording use of artifact '%s'", id.String())
+	}
```

Comment:
```suggestion
		d.config.Cache[id] = &artifactInfo{Size: size}
	} 
```

Guessing this was for debugging during implementation. Doesn't seem useful to have in the log going forward and would be pretty spammy.